lib/imgix/rails/tag.rb

```diff
@@ -31,127 +31,43 @@ def srcset(url_params: @url_params, widths: @widths)
-  GALAXY_NOTE_3 = { css_width: 360, dpr: 3 }
-  GALAXY_NOTE_3_LANDSCAPE = { css_width: 640, dpr: 3 }
-  GALAXY_NOTE_4 = { css_width: 360, dpr: 4 }
-  GALAXY_NOTE_4_LANDSCAPE = { css_width: 640, dpr: 4 }
-
-  # Tablets
-  IPAD = { css_width: 768, dpr: 1 };
-  IPAD_LANDSCAPE = { css_width: 1024, dpr: 1 };
-  IPAD_3 = { css_width: 768, dpr: 2 };
-  IPAD_3_LANDSCAPE = { css_width: 1024, dpr: 2 };
-  IPAD_PRO = { css_width: 1024, dpr: 2 };
-  IPAD_PRO_LANDSCAPE = { css_width: 1366, dpr: 2 };
-
-  BOOTSTRAP_SM = { css_width: 576, dpr: 1 }
-  BOOTSTRAP_MD = { css_width: 720, dpr: 1 }
-  BOOTSTRAP_LG = { css_width: 940, dpr: 1 }
-  BOOTSTRAP_XL = { css_width: 1140, dpr: 1 }
-
-  def devices
-    phones + phablets + tablets + bootstrap_breaks
-  end
-
-  def bootstrap_breaks
-    breaks = [
-      BOOTSTRAP_SM,
-      BOOTSTRAP_MD,
-      BOOTSTRAP_LG,
-      BOOTSTRAP_XL
-    ]
+  INCREMENT_PRECENTAGE = 8
 
-    breaks + breaks.map { |b| b[:dpr] = 2; b }
-  end
+  @@standard_widths = nil
 
-  def phones
-    [
-      IPHONE,
-      IPHONE_4,
-      IPHONE_6,
-      LG_G3
-    ]
-  end
+  def standard_widths
+    return @@standard_widths if @@standard_widths
 
-  def phablets
-    [
-      IPHONE_6_PLUS,
-      IPHONE_6_PLUS_LANDSCAPE,
-      MOTO_NEXUS_6,
-      MOTO_NEXUS_6_LANDSCAPE,
-      LUMIA_1520,
-      LUMIA_1520_LANDSCAPE,
-      GALAXY_NOTE_3,
-      GALAXY_NOTE_3_LANDSCAPE,
-      GALAXY_NOTE_4,
-      GALAXY_NOTE_4_LANDSCAPE
-    ]
-  end
+    prev = MINIMUM_SCREEN_WIDTH
+    @@standard_widths = []
+    while prev < MAXIMUM_SCREEN_WIDTH do
+      @@standard_widths.append(2 * (prev/2).round)   # Ensure widths are even
+      prev = prev * (1 + (INCREMENT_PRECENTAGE/100.0)*2.0)
+    end
+    @@standard_widths.push(MAXIMUM_SCREEN_WIDTH)
+    @@standard_widths.freeze
 
-  def tablets
-    [
-      IPAD,
-      IPAD_LANDSCAPE,
-      IPAD_3,
-      IPAD_3_LANDSCAPE,
-      IPAD_PRO,
-      IPAD_PRO_LANDSCAPE
-    ]
+    @@standard_widths
   end
 
   # Return the widths to generate given the input `sizes`
   # attribute.
   #
   # @return {Array} An array of {Fixnum} instances representing the unique `srcset` URLs to generate.
   def target_widths
-    min_screen_width_required = @tag_options[:min_width] || SCREEN_STEP
-    max_screen_width_required = @tag_options[:max_width] || MAXIMUM_SCREEN_WIDTH
-
-    widths = (device_widths + screen_widths).select do |w|
-      w <= max_screen_width_required && w >= min_screen_width_required
-    end.compact.uniq.sort
-
-    # Add exact widths for 1x, 2x, and 3x devices
-    if @url_params[:w]
-      widths.push(@url_params[:w], @url_params[:w] * 2, @url_params[:w] * 3)
+    min_width = @tag_options[:min_width]
+    max_width = @tag_options[:max_width]
+    if min_width || max_width
+      min_width = min_width || MINIMUM_SCREEN_WIDTH
+      max_width = max_width || MAXIMUM_SCREEN_WIDTH
+      widths = standard_widths.select { |w| min_width <= w && w <= max_width }
+    else
```

Comment:
We decided that we also want users to be able to supply an array of widths to completely override our algorithm, so that would probably go here.

Reply:
The library already has support for widths override. Check https://github.com/imgix/imgix-rails/pull/59/files/be7d8b37c0331e4f911e4b623361b58699bdf54d#diff-c941d27cde73ee56e4f362789fc8cbdfR18 .

Reply:
Oh! Nice.

Reply:
One caveat: the widths override isn't global. It needs to be specified with every `ix_image_tag`, `ix_picture_tag` usage.